src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLRouteTypeImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.opentripplanner.ext.legacygraphqlapi.datafetchers;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.stream.Collectors;
+import org.opentripplanner.ext.legacygraphqlapi.LegacyGraphQLRequestContext;
+import org.opentripplanner.ext.legacygraphqlapi.generated.LegacyGraphQLDataFetchers;
+import org.opentripplanner.ext.legacygraphqlapi.model.LegacyGraphQLRouteTypeModel;
+import org.opentripplanner.model.Agency;
+import org.opentripplanner.model.Route;
+import org.opentripplanner.routing.RoutingService;
+
+public class LegacyGraphQLRouteTypeImpl
+        implements LegacyGraphQLDataFetchers.LegacyGraphQLRouteType {
+
+    @Override
+    public DataFetcher<Agency> agency() {
+        return environment -> getSource(environment).getAgency();
+    }
+
+    @Override
+    public DataFetcher<Integer> routeType() {
+        return environment -> getSource(environment).getRouteType();
+    }
+
+    @Override
+    public DataFetcher<Iterable<Route>> routes() {
+        return environment -> {
+            Agency agency = getSource(environment).getAgency();
+            return getRoutingService(environment).getAllRoutes()
+                    .stream()
+                    .filter(route ->
+                            route.getId().getFeedId().equals(getSource(environment).getFeedId())
```

Comment:
I would not use multi line filter functions. I'd extract a method or use `Predicate` as they are a lot more readable.